### PR TITLE
Feature/trade

### PR DESCRIPTION
**Pull request by Ev-Che**
Реализовал все требуемые эндпоинты. Для Trade реализовал бизнес-логику поиска самых выгодных оферов. Тесты написал.

### CODE REVIEW

File: api/urls.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    # JWT Auth
+    path('auth/', include('djoser.urls')),
+    path('auth/', include('djoser.urls.jwt')),
+
+    # Apps
+    path('offers/', include('offer.urls'), name='offer'),
+    path('container/', include('stock_container.urls'), name='stock_container'),
+    path('items/', include('stock_management.urls'), name='stock_management'),
+    path('trades/', include('trade.urls'), name='trade'),
+]
```
Comment (PavelMartysiuk) [inline api/urls.py]: use routers

File: offer/services.py [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+
+
+def are_there_enough_free_stocks(user, stock, entry_quantity):
+    stocks_in_different_sell_offers = count_stocks_in_sell_offers(
+        user=user, stock=stock)
+    quantity_in_inventory = Inventory.objects.get(
+        user=user, stock=stock).quantity
+
+    return entry_quantity <= (quantity_in_inventory -
+                              stocks_in_different_sell_offers)
```
Comment (PavelMartysiuk) [inline offer/services.py]: add class and static methods

File: offer/views.py [hunk 3]
```diff
@@ -2,10 +2,34 @@
...
+from .permissions import IsOwner
+from . import serializers
 
 
 class OfferViewSet(viewsets.ModelViewSet):
-    serializer_class = OfferSerializer
-    queryset = Offer.objects.all()
-    permission_classes = [IsAuthenticated]
+    queryset = Offer.objects.all().select_related('stock', 'user')
+
```
Comment (PavelMartysiuk) [inline offer/views.py]: Does all serializers have nested serializer for stock and user?

File: trade/services/trade_service.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+
+class Trader:
+
+    def make_a_trade(self) -> None:
+        """Creates Trades."""
+
+        buy_offer = self._get_random_buy_offer()
+        print(f'buy offer: {buy_offer}')
+
```
Comment (PavelMartysiuk) [inline trade/services/trade_service.py:23]: add loguru for logs

**Review by PavelMartysiuk — CHANGES_REQUESTED** [review]

File: api/urls.py (hunk 1 above)
Comment (Ev-Che) [reply api/urls.py]: Using routers now.
https://github.com/Ev-Che/trade-platform/blob/feature/trade/trade_platform/urls.py

File: offer/services.py (hunk 2 above)
Comment (Ev-Che) [reply offer/services.py]: corrected

File: offer/views.py (hunk 3 above)
Comment (Ev-Che) [reply offer/views.py]: Added a queryset selection depending on the action

File: trade/services/trade_service.py (hunk 4 above)
Comment (Ev-Che) [reply trade/services/trade_service.py:23]: corrected

**Pull request merged**